### DISCUSSION TITLE
Extending a component

### DISCUSSION THREAD

**Discussion by b00f** [vuepress-theme-hope/vuepress-theme-hope#582] [Q&A] [answered]
Is it possible to etend only one component. For example extending: `components/Theme/ThemeOptions1`

**Comment by Mister-Hope** [comment, marked as answer]
It's. See https://v1.vuepress.vuejs.org/theme/inheritance.html#usage.

Feel free to ask if you meet issues while extending it.

**Reply by b00f** [reply]
I would like to extend only one of the hope's components and keep others untouched. I did something like this, but it doesn't work. 

```
module.exports = {
  extend: 'vuepress-theme-hope'
}
```

**Reply by Mister-Hope** [reply]
For a theme to work properly, I think you need to at least provide `layouts/Layout.vue`. Replace all my `@theme` with `@parent-theme`

**Reply by Mister-Hope** [reply]
This folder is using theme-extending, hope it helps you.

https://github.com/vuepress-theme-hope/vuepress-theme-hope/tree/v1/packages/comment/__tests__/demo

You may need to overload all the require chain on `<ThemeOption />` component

**Comment by Mister-Hope** [comment]
BTW, is there any feature you want, but I did not provide in `<ThemeOption />`?

**Reply by b00f** [reply]
I believe these items can help users for more customization: 

- User can decide about the color of boxes in the main page.
- Simple dark/light switch instead of current ThemeOption menu.
- User can set `DefaultColor` in styles for changing default color.

**Reply by Mister-Hope** [reply]
For the third one, do you mean changing like this? https://mrhope.site

You can do this by setting the $accentColor var in palette.styl. Related docs are at https://vuepress-theme-hope.github.io/config/stylus/

There are a lot you can config.

**Reply by Mister-Hope** [reply]
For the first one, it's hard to support, but I will ship a "pure feature" in future to diable all the fancy styles. See #438

**Reply by Mister-Hope** [reply]
The second one is also hard to support, because I think a switch with auto is better than "light/dark" and it's taking too much space.

Also I am providing themeColor and full screen feature, so a popup is provided.

It's possible to ship, but hard and a little worthless where the correct component is directly rendered at navbar  when only one of darkmode, themeColor or fullScreen is enabled.

**Reply by b00f** [reply]
I also look for a solution to disable the fancy motions when a new page is loading. I want something super simple. I don't know which plugin is doing that!

**Reply by Mister-Hope** [reply]
`MyTransition` component is doing that.

**Reply by b00f** [reply]
Is it possible to disable `MyTransition` component?

**Reply by Mister-Hope** [reply]
Currenly no, it you want to do that, you need to overide all the components using it. I would consider bringing pure mode these days. You can just wait.

**Comment by b00f** [comment]
One more thing, `Edit on Github` in https://vuepress-theme-hope.github.io/ doesn't work. 
Algolia search also doesn't work for me ;(

**Reply by Mister-Hope** [reply]
Algolia search is broken because the link changed (from `vuepress-theme.mrhope.site`), and algolia has not responded with my change request now.

**Reply by Mister-Hope** [reply]
And thanks for the edit link report! I had changed the default branch to v1 but I didn't config the docs. Thanks!

**Reply by Mister-Hope** [reply]
Edit on Github should work now.
